src/common/selector/create-shallow-equal-selector.spec.ts

```diff
@@ -0,0 +1,60 @@
+import createShallowEqualSelector from './create-shallow-equal-selector';
+
+describe('createShallowEqualSelector()', () => {
+    interface TestState {
+        items: Array<{ id: number }>;
+        messages: string[];
+    }
+
+    it('creates selector that does shallow comparison instead of strict comparison', () => {
```

Comment:
wouldn't the test behave the same if you were doing strict comparison?

Reply:
Good question. It would be different because if you look at the following:

```ts
const result = selector({ messages: ['a', 'b'], items: [] });
const result2 = selector({ messages: ['a', 'b'], items: [] });
```

I'm actually passing in a new array instance for each call. Therefore, it won't hit the cache if the selector is configured to use strict comparison.